operator/internal/validation/rulerconfig.go

```diff
@@ -0,0 +1,110 @@
+package validation
+
+import (
+	"context"
+	"fmt"
+
+	lokiv1beta1 "github.com/grafana/loki/operator/apis/loki/v1beta1"
+
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/util/validation/field"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/webhook/admission"
+)
+
+var _ admission.CustomValidator = &RulerConfigValidator{}
+
+// RulerConfigValidator implements a custom validator for RulerConfig resources.
+type RulerConfigValidator struct {
+	ExtendedValidator func(context.Context, *lokiv1beta1.RulerConfig) field.ErrorList
```

Comment:
As far as I can see we do not need the `ExtendedValidator` yet. Should we remove it until we actually need it?

Reply:
removed